File: filters/src/main/scala/com/sksamuel/scrimage/filter/PipelineFilter.scala
package com.sksamuel.scrimage.filter

import com.sksamuel.scrimage.{Image, Filter}

/** @author Stephen Samuel
  *
  *         Applies several filters at once in sequential order
  *
  * */
class PipelineFilter(filters: Filter*) extends Filter {
  def apply(image: Image) = filters.foldLeft(image)((image, filter) => image.filter(filter))
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/PipelineFilter.java
package com.sksamuel.scrimage;

import java.util.Collection;

public class PipelineFilter implements Filter {

    private final Collection<Filter> filters;

    public PipelineFilter(Collection<Filter> filters) {
        this.filters = filters;
    }

    @Override
    public void apply(ImmutableImage image) {
        for (Filter filter : filters) {
            filter.apply(image);
        }
    }
}
